applied-crypto-examples/src/encrypted_zksnark.rs

```diff
@@ -0,0 +1,254 @@
+//! An example of naive implementations of zksnarks for tutorial purposes
+use curve25519_dalek::{
+    constants::RISTRETTO_BASEPOINT_POINT, ristretto::RistrettoPoint, scalar::Scalar,
+};
+const G: RistrettoPoint = RISTRETTO_BASEPOINT_POINT;
+
+/// Single root of a polynomial
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct Root {
+    // a in ax+b
+    a: Scalar,
+    // b in ax+b
+    b: Scalar,
+}
+
+impl Root {
+    /// Create a new root from i64 values
+    pub fn new(a: i64, b: i64) -> Result<Self, Error> {
```

Comment:
Optimally we should change this to something else? Like just initialize with scalars or have any integers be able to translate into scalars/create "from" implementations. For the tutorial maybe this okay, but I'd like to see a follow-on PR that changes this.

Reply:
Yeah, probably just scalars and have a from function that translates between i64. If this turns into a productoin version, we might make it more robust.